cyral/resource_cyral_repository_conf_auth.go

```diff
@@ -204,6 +210,13 @@ func repositoryConfAuthResourceSchemaV0() *schema.Resource {
 				Optional:    true,
 				Default:     defaultRepoTLS,
 			},
+			"auth_type": {
+				Description: "Authentication type for this repository. 0 for Opaque Token, " +
+					"1 for AWS IAM. Defaults to 0",
+				Type:     schema.TypeInt,
```

Comment:
For enum fields, we usually define them as `schema.TypeString`. I believe this makes the valid values more explicit and self-explanatory in some sense. We could do the same here and also use the `supportedTypesMarkdown` function in the field description to explicitly list the valid values for this field.